### PR TITLE
[ao][sparsity] make sparsity compose with PTQ convert

### PR DESCRIPTION
**Pull request by HDCharles**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #74848
* #74847
* __->__ #74846

Summary: This PR primarily allows the PTQ convert function to work with
parametrized modules. Given that the parametrized weight is what is used
by default in convert, as long as sparsifier.step() has already been
called, the converted model will use the sparisified weights. There is
currently no way to handle things if sparsifier.step() has not been
called. Lastly, added the has_no_children_ignoring_parametrizations function because
parametrized modules no longer look like leaves due to the
parametrizations module attached to them

Test Plan: python test/test_ao_sparsity.py TestComposability

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D35240275](https://our.internmc.facebook.com/intern/diff/D35240275)

### CODE REVIEW

**Comment by HDCharles** [comment]
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35240275).

File: torch/nn/quantized/modules/linear.py [hunk 1]
```diff
@@ -8,6 +8,8 @@
 from torch.nn.utils.fusion import fuse_linear_bn_weights
 from typing import Optional
 
+from torch.nn.utils.parametrize import is_parametrized
```
Comment (vkuzo) [inline torch/nn/quantized/modules/linear.py:11]: nit: fix lint

**Review by vkuzo — APPROVED** [review]

**Comment by HDCharles** [comment]
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35240275).

**Comment by HDCharles** [comment]
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35240275).

**Comment by HDCharles** [comment]
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35240275).

File: test/ao/sparsity/test_composability.py [hunk 2]
```diff
@@ -92,3 +95,27 @@ def test_s_prep_before_q_prep(self):
         )
         self.assertTrue(isinstance(mod[5], torch.nn.quantized.Linear))
         self.assertEqual(mod(torch.randn(1, 4, 4, 4)).shape, torch.Size([1, 4, 4, 4]))
+
+    def test_convert_without_squash_mask(self):
```
Comment (jerryzh168) [inline test/ao/sparsity/test_composability.py:99]: nit: same more this, I feel these tests are a bit complicated so we should have more comments on the context

**Comment by HDCharles** [comment]
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35240275).

**Comment by HDCharles** [comment]
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35240275).

**Comment by HDCharles** [comment]
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35240275).

**Comment by HDCharles** [comment]
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35240275).

**Pull request closed without merging**